### DISCUSSION TITLE
Menu isn't sorted correctly

### DISCUSSION THREAD

**Discussion by geofftaylor** [ttscoff/bunch-docs#9] [Q&A] [answered]
In Bunch 1.4.0 beta 67 (macOS 10.15.7), the menu isn't sorted correctly. (It also wasn't correct in beta 66, but I just got around to posting.) For example, here's my "Blog" bunch's frontmatter:

```
---
title: ✏️Blog
menu order: 1
---
```

Here's a screenshot of the menu:

<img width="243" alt="Bunch menu" src="https://user-images.githubusercontent.com/518246/107291036-84472c80-6a35-11eb-80e7-67580030a62d.png">

All of the bunches have a `menu order` key, but none of them are in the correct position (except "Login," which has `menu order: 100`). When I quit Bunch and open it again, the menu order is completely different (except "Login," which always appears at the end).

**Comment by ttscoff** [comment]
I've noticed something strange going on with my own bunches as well. I'm investigating. I'll open this as an issue as well.

**Reply by ttscoff** [reply]
Issue #12 opened.

**Reply by ttscoff** [reply]
Fixed for next update

**Reply by geofftaylor** [reply]
Thanks!

**Reply by maxoriggi** [reply]
Played a bit with beta 68 and it still has sorting problems, it sorts differently than beta 67, it sorts some bunches correctly, but  a few of them are still off. 
Changing menu order affects the result, but I can't find a pattern to try and understand what's wrong.

**Comment by ttscoff** [comment]
Found the issue, it's fixed for real for the next update.

**Comment by ttscoff** [comment, marked as answer]
Was failing to convert to integer values before doing comparisons, so where it got it right it was partly just getting lucky. Beta 69 is out now and should fix it, please let me know if you still see issues.

**Reply by maxoriggi** [reply]
Looks good now!

**Reply by geofftaylor** [reply]
It's working for me as well.

**Reply by ttscoff** [reply]
This is great news. The whole "sort numbers at the beginning, sort alphanumeric for duplicates, sort alphanumeric for unnumbered, sort high numbers at the end" thing proved more challenging than I would have suspected.

**Reply by ttscoff** [reply]
@geofftaylor I'm still figuring out how these discussions work. I can't mark this response as "Answered" the way I could close a ticket. Are you, as the OP, able to do so?

**Comment by ttscoff** [comment]
@geofftaylor so what does your Login bunch do?

**Reply by geofftaylor** [reply]
It just opens a few apps that I always have open, sends a keyboard shortcut to restore the previous session in Firefox, and sets the window arrangement with Moom. It used to be called by a startup file, but now I just have `startup: true` in the frontmatter (and `only opens: true` as of the latest beta).

**Reply by ttscoff** [reply]
Oh, it's "on login." I thought maybe it logged you into something else :).

**Reply by geofftaylor** [reply]
Nothing that clever. :) Just a convenient way to get up and running after a reboot.
